internal/mode/static/nginx/config/servers_template.go

```diff
@@ -22,6 +22,7 @@ server {
     listen {{ $s.Port }} ssl;
     ssl_certificate {{ $s.SSL.Certificate }};
     ssl_certificate_key {{ $s.SSL.CertificateKey }};
+    proxy_ssl_server_name on;
```

Comment:
This belongs below (near line 63) with the other directives that are only enabled when a BackendTLSPolicy exists.